### ISSUE TITLE
Included Resources only included in first request

### ISSUE THREAD

**Issue by mdcooper** [codecutout/JsonApiSerializer#37]
With .Net Core and a default formatter set as shown in #35, I am getting the included resources only on the first request. Additional requests to the same endpoint do not have the included resources in the returned json. I am eagerly loading the related entities and they do indeed appear in the query response before serialization every time. Is this expected behavior? Do I have something misconfigured?

Thanks!

**Comment by alex-davies** [comment]
Thanks for pointing this out.

I think I see where the issue is. Looks like the default behaviour on the formatters is to cache the Serializer (https://github.com/aspnet/Mvc/blob/dev/src/Microsoft.AspNetCore.Mvc.Formatters.Json/JsonOutputFormatter.cs#L122-L130). However the json.net `ReferenceResolver` is created once per serializer. The `JsonApiSerializer` library makes heavy use of a customer `ReferenceResolver` to track included items. So issue is that it is not creating a new `ReferenceResolver` for each serialization.

I need to investigate a bit further to either make this library work with a single `ReferenceResolver` or reproduce a similar issue with vanilla json.net and raise a bug against the MVC framework .

But until I work out an in library fix, you can apply a quickfix by modifing the `JsonOutputSerializer` to not cache the serializer

```csharp
    public class JsonApiOutputFormatter : JsonOutputFormatter
    {
        private JsonApiSerializerSettings _serializerSettings;

        public JsonApiOutputFormatter() : this(new JsonApiSerializerSettings(), ArrayPool<Char>.Shared)
        {

        }

        public JsonApiOutputFormatter(ArrayPool<Char> charPool) : this(new JsonApiSerializerSettings(), charPool)
        {
      
        }

        private JsonApiOutputFormatter(JsonApiSerializerSettings jsonSerializerSettings, ArrayPool<Char> charPool) : base(jsonSerializerSettings, charPool)
        {
            _serializerSettings = jsonSerializerSettings; 
        }

        protected override JsonSerializer CreateJsonSerializer()
        {
            return JsonSerializer.Create(_serializerSettings);
        }
    }
```

**Comment by alex-davies** [comment]
This should be fixed in 1.2.0

After a largish refactor I have removed the `ReferenceResolver` and using the `JsonReader`/`JsonWriter` instance to pin any per-serialization data to

So now MVC can cache the serializer ,os long as it passes down a new JsonReader/JsonWriter for each serialization (which it does) it should work
